dali/pipeline/util/thread_pool.cc

```diff
@@ -215,8 +235,8 @@ void ThreadPool::ThreadMain(int thread_id, int device_id, bool set_affinity,
       //                                  lock.lock()
       //                                  continue execution
 
-      lock.lock();
-      lock.unlock();
+      std::unique_lock<std::mutex> lock2(completed_mutex_);
+      lock2.unlock();
```

Comment:
Nit: Just use a tiny scope with a lock_guard instead of a more complex unique_lock. Or you can even do:
```suggestion
(void)std::lock_gruard(completed_mutex_);
```